src/components/ContactsModal/Body.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { ChangeEvent, useEffect, useRef, useState } from 'react';
+import styles from './Body.module.scss';
+import ContactList from "../ContactList/ContactList";
+import { useAppDispatch, useAppSelector } from "../../store/hooks";
+import { getContacts } from "../../store/features/contacts/contactsSlice";
+
+interface Props {
+    country: '' | 'us',
+    onlyEven: boolean,
+}
+
+const ContactsModalBody = (props: Props) => {
+    const { country, onlyEven } = props;
+    const [query, setQuery] = useState('');
+    const [page, setPage] = useState(1);
+    const [inputTimeout, setInputTimeout] = useState<any>(null);
+    const dispatch = useAppDispatch();
+    const { status, data } = useAppSelector(state => state.contacts);
+    const { contacts, next } = data;
+    const listRef = useRef<any>(null);
+    const firstRender = useRef(true);
+    const isLoading = status === 'loading';
+
+    useEffect(() => {
+        if ((page > 1 && !next) || isLoading) return;
+
+        dispatch(getContacts({ country, query, page, onlyEven: onlyEven ? 1 : 0 }))
+    }, [page])
+
+    useEffect(() => {
+        if (firstRender.current) {
```

Comment:
de ce ai nevoie sa verifici daca este sau nu prima randare?

Reply:
pentru ca e al doile useEffect in care folosesc dispatch(getContacts) si s-ar face request ul de doua ori la mount